README.md

```diff
@@ -131,9 +131,124 @@ You can add your customized Python code to process your input and output data. T
+
+    import json
+    import requests
+    
+    
+    def handler(data, context):
+        """Handle request.
+        Args:
+            data (obj): the request data
+            context (Context): an object containing request and configuration details
+        Returns:
+            (bytes, string): data to return to client, (optional) response content type
+        """
+        processed_input = _process_input(data, context)
+        response = requests.post(context.rest_uri, data=processed_input)
+        return _process_output(response, context)
+    
+    
+    def _process_input(data, context):
+        if context.request_content_type == 'application/json':
+            # pass through json (assumes it's correctly formed)
+            d = data.read().decode('utf-8')
+            return d if len(d) else ''
+    
+        if context.request_content_type == 'text/csv':
+            # very simple csv handler
+            return json.dumps({
+                'instances': [float(x) for x in data.read().decode('utf-8').split(',')]
+            })
+    
+        raise ValueError('{{"error": "unsupported content type {}"}}'.format(
+            context.request_content_type or "unknown"))
+
+
+    def _process_output(data, context):
+        if data.status_code != 200:
+            raise ValueError(data.content.decode('utf-8'))
+    
+        response_content_type = context.accept_header
+        prediction = data.content
+        return prediction, response_content_type
+    
 You can also bring in external dependencies to help with your data processing. There are 2 ways to do this:
-1. If your model archive contains `code/requirements.txt`, the container will install the python dependencies at runtime using `pip install -r`.
-2. If you have pre-installed modules under `code/lib` directory in your model archive, the container will add the modules to the Python path. Note that if both `code/lib` and `code/requirements.txt` are present in the model archive, the `requirements.txt` will be ignored.
+1. If your model archive contains `code/requirements.txt`, the container will install the Python dependencies at runtime using `pip install -r`.
+2. If you are working in a network-isolation situation, you may want to put pre-downloaded dependencies under `code/lib` directory in your model archive, the container will then add the modules to the Python path. Note that if both `code/lib` and `code/requirements.txt` are present in the model archive, the `requirements.txt` will be ignored.
+
+Your model directory structure may look like this if you are using `requirements.txt`:
+
+    models
+        |__model1
+        |__model2
+        |__code
+            |__inference.py
+            |__requirements.txt
+
+Your model directory structure may look like this if you have downloaded modules under `code/lib`:
+
+    models
+        |__model1
+        |__model2
+        |__code
+            |__lib
+                |__external_module
+            |__inference.py
 
```

Comment:
Could you add instructions on making your own model.tar.gz, for non-Python SDK users? Could you clarify that the SageMaker Model entity has to point to this model.tar.gz in S3 as its model data URL?

Reply:
done.